### PR TITLE
Registry integration test: Ignore

### LINKED ISSUE #782: Registry integration test: Improve test_ignore_registry.py
Hello team!

The test `test_ignore_registry` found in `test_ignore_registry.py` only search the message `Ignoring '%s' '%s' due to sregex '%s'` to verify the option `registry_ignore` works. I think the test has the following failures:

- Only check if the registry is ignored when the regex is used. There must be a callback to check message `"(6204): Ignoring '%s' '%s' due to '%s'"`
- It doesn't modify the registries monitored, and the subkeys of registries monitored. It does we can't detect some bugs like reported in issue [5251](https://github.com/wazuh/wazuh/issues/5251)

Best regards,
Eva

### PR DESCRIPTION
**Pull request by antoniomanuelfr**
Hello team,
This PR aims to add a new test for the Windows registry integration test.
Closes #782

# Tests logic
 ### First case
- Monitor two keys: One with `arch="64bit` and another with `arch="both"` and using the `registry_ignore` option (using the path to the ignored key and using the regex option) to ignore certain keys
- It creates keys and checks that all the events are generated if the key is not ignored. In case that the key is ignored, the test checks that the message `Ignoring 'key' 'HKEY_........` appears.
### Second case
- Monitor two keys: One with `arch="64bit` and another with `arch="both"` and using the `registry_ignore_value` option (using the path of the value and the regex option) to ignore certain values.
- It creates values and checks that all the events are generated if the value is not ignored. In case that the value is ignored, the test checks that the message `Ignoring 'value' 'HKEY_........` appears.
# Tests checks

- [X] Proven that tests **pass** when they have to pass
- [X] Proven that tests **fail** when they have to fail

## Windows

- When the expected behavior occurs:

```
python -m pytest -vv test_ignore\
============================= test session starts 
metadata: {'Python': '3.7.3', 'Platform': 'Windows-7-6.1.7601-SP1'}
collected 36 items

[  2%] test_ignore_registry.py PASSED
[  5%] test_ignore_registry.py PASSED
[  8%] test_ignore_registry.py PASSED
[ 11%] test_ignore_registry.py PASSED
[ 13%] test_ignore_registry.py PASSED
[ 16%] test_ignore_registry.py PASSED
[ 19%] test_ignore_registry.py PASSED
[ 22%] test_ignore_registry.py PASSED
[ 25%] test_ignore_registry.py SKIPPED
[ 27%] test_ignore_registry.py SKIPPED
[ 30%] test_ignore_registry.py SKIPPED
[ 33%] test_ignore_registry.py SKIPPED
[ 36%] test_ignore_registry.py SKIPPED
[ 38%] test_ignore_registry.py SKIPPED
[ 41%] test_ignore_registry.py SKIPPED
[ 44%] test_ignore_registry.py SKIPPED
[ 47%] test_ignore_registry.py SKIPPED
[ 50%] test_ignore_registry.py SKIPPED
[ 52%] test_ignore_registry.py SKIPPED
[ 55%] test_ignore_registry.py SKIPPED
[ 58%] test_ignore_registry.py SKIPPED
[ 61%] test_ignore_registry.py SKIPPED
[ 63%] test_ignore_registry.py SKIPPED
[ 66%] test_ignore_registry.py SKIPPED
[ 69%] test_ignore_registry.py SKIPPED
[ 72%] test_ignore_registry.py SKIPPED
[ 75%] test_ignore_registry.py PASSED
[ 77%] test_ignore_registry.py PASSED
[ 80%] test_ignore_registry.py PASSED
[ 83%] test_ignore_registry.py PASSED
[ 86%] test_ignore_registry.py PASSED
[ 88%] test_ignore_registry.py PASSED
[ 91%] test_ignore_registry.py PASSED
[ 94%] test_ignore_registry.py PASSED
[ 97%] test_ignore_registry.py PASSED
[100%]
============== 18 passed, 18 skipped in 37.11s ===================
```
Best regards.

### CODE REVIEW

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 1]
```diff
@@ -8,66 +8,45 @@
 
 import pytest
 
-from wazuh_testing.fim import LOG_FILE_PATH, callback_ignore, create_registry, delete_registry, registry_parser
+from wazuh_testing.fim import LOG_FILE_PATH, callback_ignore, create_registry, delete_registry, registry_parser, \
+        KEY_WOW64_32KEY, KEY_WOW64_64KEY, generate_params, callback_detect_event, check_time_travel, modify_registry_value
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:12]: `delete_registry` is imported but unused. Also, line 12 exceeds 120 characters.
```suggestion
from wazuh_testing.fim import LOG_FILE_PATH, callback_ignore, create_registry, registry_parser, KEY_WOW64_32KEY, \
    KEY_WOW64_64KEY, generate_params, callback_detect_event, check_time_travel, modify_registry_value
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 2]
```diff
@@ -8,66 +8,45 @@
...
 from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
 from wazuh_testing.tools.monitoring import FileMonitor
 from wazuh_testing.tools.time import TimeMachine
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:15]: `TimeMachine` is imported but unused.
```suggestion
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 3]
```diff
@@ -8,66 +8,45 @@
...
-        else:
-            arch = win32con.KEY_WOW64_32KEY if keys[1] == '32' else win32con.KEY_WOW64_64KEY
-            delete_registry(registry_parser[key], keys[0], arch)
+reg1, reg2 = test_regs
 
+# Configurations
+conf_params = {'WINDOWS_REGISTRY_1':reg1,
+               'WINDOWS_REGISTRY_2':reg2,
+               'REGISTRY_IGNORE_1': os.path.join(reg1, ignore_key),
+               'REGISTRY_IGNORE_2' : os.path.join(reg2, ignore_key),
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:40]: Whitespace before `:`.
https://www.flake8rules.com/rules/E203.html
```suggestion
               'REGISTRY_IGNORE_2': os.path.join(reg2, ignore_key),
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 4]
```diff
@@ -8,66 +8,45 @@
...
+               'REGISTRY_IGNORE_REGEX': ignore_regex,
+               'VALUE_IGNORE_1' : os.path.join(reg1, ignore_value),
+               'VALUE_IGNORE_2' : os.path.join(reg2, ignore_value),
+               'VALUE_IGNORE_REGEX' : ignore_value_regex
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:44]: Whitespace before :.
```suggestion
               'VALUE_IGNORE_1': os.path.join(reg1, ignore_value),
               'VALUE_IGNORE_2': os.path.join(reg2, ignore_value),
               'VALUE_IGNORE_REGEX': ignore_value_regex
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 5]
```diff
@@ -76,11 +55,19 @@ def get_configuration(request):
...
+    (key, subkey_1, KEY_WOW64_64KEY, "regex_ignored_key", False, {'ignore_registry_key'}),
+    (key, subkey_1, KEY_WOW64_32KEY, "regex_ignored_key", False, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "some_key", True, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, ignore_key, False, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "regex_ignored_key", False, {'ignore_registry_key'})
 ])
-def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuration,
-                         configure_environment, restart_syscheckd, wait_for_initial_scan):
+def test_ignore_registry_key(root_key, registry, subkey, arch, triggers_event, tags_to_apply,
+                         get_configuration, configure_environment, restart_syscheckd, wait_for_fim_start):
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:70]: This line is under-indented.
```suggestion
                             get_configuration, configure_environment, restart_syscheckd, wait_for_fim_start):
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 6]
```diff
@@ -76,11 +55,19 @@ def get_configuration(request):
...
+    (key, subkey_1, KEY_WOW64_32KEY, ignore_key, False, {'ignore_registry_key'}),
+    (key, subkey_1, KEY_WOW64_64KEY, "regex_ignored_key", False, {'ignore_registry_key'}),
+    (key, subkey_1, KEY_WOW64_32KEY, "regex_ignored_key", False, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "some_key", True, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, ignore_key, False, {'ignore_registry_key'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "regex_ignored_key", False, {'ignore_registry_key'})
 ])
-def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuration,
-                         configure_environment, restart_syscheckd, wait_for_initial_scan):
+def test_ignore_registry_key(root_key, registry, subkey, arch, triggers_event, tags_to_apply,
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:69]: These parameters are not documented.

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 7]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+    create_registry(registry_parser[root_key], os.path.join(registry, subkey), arch)
+    # Go ahead in time to let syscheck perform a new scan
+    check_time_travel(scheduled, monitor=wazuh_log_monitor)
+
+    if triggers_event:
+        event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                    callback=callback_detect_event,
+                                    error_message='Did not receive expected '
+                                                    '"Sending FIM event: ..." event',
+                                    accum_results=2).result()
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:95]: These lines are under-indented.
```suggestion
                                        callback=callback_detect_event,
                                        error_message='Did not receive expected '
                                                      '"Sending FIM event: ..." event',
                                        accum_results=2).result()
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 8]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+
+        assert event[0]['data']['type'] == 'added', f'Wrong event type.'
+        assert event[0]['data']['path'] == os.path.join(root_key, registry, subkey), f'Wrong key path.'
+        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'wrong key arch.'
+
+        assert event[1]['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event[1]['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:103]: None of these error messages are using the format option, please remove it. By removing the `f`, line 103 will not exceed 120 characters.
Also, please capitalize the error message in line 99.
```suggestion
        assert event[0]['data']['type'] == 'added', 'Wrong event type.'
        assert event[0]['data']['path'] == os.path.join(root_key, registry, subkey), 'Wrong key path.'
        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Wrong key arch.'

        assert event[1]['data']['type'] == 'modified', 'Parent key event type not equal'
        assert event[1]['data']['path'] == os.path.join(root_key, registry), 'Wrong parent key path.'
        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Parent key arch not equal.'
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 9]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+    else:
+        # The ignore event is generated before the event of the parent key.
+        while True: # Look for the ignore event of the created key
+            ignored_key = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                                   callback=callback_ignore).result()
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:108]: This line is over-indented.
```suggestion
                                                  callback=callback_ignore).result()
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 10]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+        assert event[0]['data']['type'] == 'added', f'Wrong event type.'
+        assert event[0]['data']['path'] == os.path.join(root_key, registry, subkey), f'Wrong key path.'
+        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'wrong key arch.'
+
+        assert event[1]['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event[1]['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
+    else:
+        # The ignore event is generated before the event of the parent key.
+        while True: # Look for the ignore event of the created key
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:106]: Please, have at least two spaces before an inline comment: https://www.flake8rules.com/rules/E261.html
```suggestion
        while True:  # Look for the ignore event of the created key
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 11]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+            ignored_key = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                                   callback=callback_ignore).result()
+            if ignored_key == os.path.join(key, os.path.join(registry, subkey)):
+                break
+
+        event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            callback=callback_detect_event,
+                            error_message='Did not receive expected '
+                                            '"Sending FIM event: ..." event',
+                            accum_results=1).result()
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:116]: These lines are under-indented.
```suggestion
                                        callback=callback_detect_event,
                                        error_message='Did not receive expected '
                                                      '"Sending FIM event: ..." event',
                                        accum_results=1).result()
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 12]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+
+        assert event['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:120]: None of these error messages is using the format option, please remove the `f`.
```suggestion
        assert event['data']['type'] == 'modified', 'Parent key event type not equal'
        assert event['data']['path'] == os.path.join(root_key, registry), 'Wrong parent key path.'
        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Parent key arch not equal.'
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 13]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+    (key, subkey_1, KEY_WOW64_64KEY, "some_value", True, {'ignore_registry_value'}),
+    (key, subkey_1, KEY_WOW64_64KEY, "regex_ignored_value", False, {'ignore_registry_value'}),
+    (key, subkey_1, KEY_WOW64_32KEY, "regex_ignored_value", False, {'ignore_registry_value'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "regex_ignored_value", False, {'ignore_registry_value'}),
+    (key, subkey_2, KEY_WOW64_64KEY, "some_value", True, {'ignore_registry_value'}),
+    (key, subkey_1, KEY_WOW64_32KEY, ignore_value, False, {'ignore_registry_value'}),
+    (key, subkey_1, KEY_WOW64_64KEY, ignore_value, False, {'ignore_registry_value'}),
+    (key, subkey_2, KEY_WOW64_64KEY, ignore_value, False, {'ignore_registry_value'})
+])
+def test_ignore_registry_value(root_key, registry, arch, value, triggers_event, tags_to_apply,
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:134]: These parameters are not documented.

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 14]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+                         get_configuration, configure_environment, restart_syscheckd, wait_for_fim_start):
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:135]: This line is under-indented.
```suggestion
                               get_configuration, configure_environment, restart_syscheckd, wait_for_fim_start):
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 15]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+    # Create values
+    modify_registry_value(key_h, value, REG_SZ, "test_value")
+    # Go ahead in time to let syscheck perform a new scan
+    check_time_travel(scheduled, monitor=wazuh_log_monitor)
+
+    event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_event,
+                                error_message='Did not receive expected '
+                                                '"Sending FIM event: ..." event',
+                                accum_results=2 if triggers_event else 1).result()
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:161]: These lines are under-indented.
```suggestion
                                    callback=callback_detect_event,
                                    error_message='Did not receive expected '
                                                  '"Sending FIM event: ..." event',
                                    accum_results=2 if triggers_event else 1).result()
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 16]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+
+    if triggers_event:
+
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:164]: Please, remove this blank line.
```suggestion
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 17]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+        assert event[0]['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event[0]['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
+
+        assert event[1]['data']['type'] == 'added', f'Wrong event type.'
+        assert event[1]['data']['path'] == os.path.join(root_key, registry), f'Wrong value path.'
+        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'wrong key arch.'
+        assert event[1]['data']['value_name'] == value, f'Wrong value name'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:172]: None of these error messages are using the format option, please remove it. By removing the f, line 167 will not exceed 120 characters.
Also, please capitalize the error message in line 171.
```suggestion
        assert event[0]['data']['type'] == 'modified', 'Parent key event type not equal'
        assert event[0]['data']['path'] == os.path.join(root_key, registry), 'Wrong parent key path.'
        assert event[0]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Parent key arch not equal.'

        assert event[1]['data']['type'] == 'added', 'Wrong event type.'
        assert event[1]['data']['path'] == os.path.join(root_key, registry), 'Wrong value path.'
        assert event[1]['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Wrong key arch.'
        assert event[1]['data']['value_name'] == value, 'Wrong value name'
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 18]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:173]: Please, remove this blank line.
```suggestion
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 19]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+    else:
+        assert event['data']['type'] == 'modified', f'Parent key event type not equal'
+        assert event['data']['path'] == os.path.join(root_key, registry), f'Wrong parent key path.'
+        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', f'Parent key arch not equal.'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:177]: None of these error messages are using the format option, please remove it.
```suggestion
        assert event['data']['type'] == 'modified', 'Parent key event type not equal'
        assert event['data']['path'] == os.path.join(root_key, registry), 'Wrong parent key path.'
        assert event['data']['arch'] == '[x32]' if arch == KEY_WOW64_32KEY else '[x64]', 'Parent key arch not equal.'
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 20]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+
+        while True: # Look for the ignore event of the created value
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:179]: Please, have at least two spaces before an inline comment.
```suggestion
        while True:  # Look for the ignore event of the created value
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 21]
```diff
@@ -94,17 +81,103 @@ def test_ignore_registry(key_string, key_object, tags_to_apply, get_configuratio
...
+            ignored_value = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                                   callback=callback_ignore).result()
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:181]: This line is under-indented.
```suggestion
                                                    callback=callback_ignore).result()
```

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 22]
```diff
@@ -8,66 +8,45 @@
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:8]: `sys` and `timedelta` are imported but unused.

File: tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py [hunk 23]
```diff
@@ -8,66 +8,45 @@
...
-            delete_registry(registry_parser[key], keys[0], win32con.KEY_WOW64_32KEY)
-            delete_registry(registry_parser[key], keys[0], win32con.KEY_WOW64_64KEY)
-        else:
-            arch = win32con.KEY_WOW64_32KEY if keys[1] == '32' else win32con.KEY_WOW64_64KEY
-            delete_registry(registry_parser[key], keys[0], arch)
+reg1, reg2 = test_regs
 
+# Configurations
+conf_params = {'WINDOWS_REGISTRY_1':reg1,
+               'WINDOWS_REGISTRY_2':reg2,
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/test_ignore_registry.py:38]: Missing whitespace after `:`.
```suggestion
conf_params = {'WINDOWS_REGISTRY_1': reg1,
               'WINDOWS_REGISTRY_2': reg2,
```

File: tests/integration/test_fim/test_registry/test_ignore/data/wazuh_registry_ignore_conf.yaml [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+        - registry_ignore:
+            value: "REGISTRY_IGNORE_REGEX"
+            attributes:
+                - type: 'sregex'
+                - arch: 'both'
+        - registry_ignore:
+            value: "REGISTRY_IGNORE_REGEX"
+            attributes:
+                - type: 'sregex'
+                - arch: '64bit'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/data/wazuh_registry_ignore_conf.yaml:29]: Why adding the same regex for both architectures and for 64 bit alone?

File: tests/integration/test_fim/test_registry/test_ignore/data/wazuh_registry_ignore_conf.yaml [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+        - registry_ignore_value:
+            value: "VALUE_IGNORE_REGEX"
+            attributes:
+                - type: 'sregex'
+                - arch: 'both'
+        - registry_ignore_value:
+            value: "VALUE_IGNORE_REGEX"
+            attributes:
+                - type: 'sregex'
+                - arch: '64bit'
```
Comment (jesusjimsa) [inline tests/integration/test_fim/test_registry/test_ignore/data/wazuh_registry_ignore_conf.yaml:65]: Why adding the same regex for both architectures and for 64 bit alone?

**Review by jesusjimsa — CHANGES_REQUESTED** [review]
This test is failing.
[test_registry.zip](https://github.com/wazuh/wazuh-qa/files/5452998/test_registry.zip)

**Review by jesusjimsa — APPROVED** [review]
LGTM!

**Review by bah07 — APPROVED** [review]
LGTM!

**Pull request merged**